@here/harp-mapview/lib/text/TextElementGroupState.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TextElement } from "./TextElement";
+import { TextElementGroup } from "./TextElementGroup";
+import { TextElementState } from "./TextElementState";
+
+/**
+ * Type of functions used to do early rejection of elements during group state creation or update.
+ * @param textElement The text element to check.
+ * @param lastFrameNumber Last frame the element was processed for rendering if available, otherwise
+ * `undefined`.
+ * @returns `undefined` if element was rejected, otherwise its current view distance.
+ */
+export type TextElementFilter = (
+    textElement: TextElement,
+    lastFrameNumber?: number
+) => number | undefined;
+
+/**
+ * `TextElementGroupState` keeps the state of a text element group while it's being rendered.
+ */
+export class TextElementGroupState {
+    // Sorted list of element states belonging to the group state. It does NOT have the same order
+    // as their elements in the group.
+    private m_textElementStates: TextElementState[];
+    private m_visited: boolean = false;
+    private m_needsSorting: boolean = false;
```

Comment:
The constructor invalidates the sorting whenever a TextElementState is created with a valid viewDistance. By default is false because you might create a text element group state where all text element states are not initialized (e.g. all are further than the max view distance). Then you don't need to sort by distance.